Subject: [PATCH] Fix for NAT core on nonlinear skbs
X-BK-ChangeSetKey: rusty@rustcorp.com.au[torvalds]|ChangeSet|20050105041500|12440
Date:	Wed, 05 Jan 2005 17:10:51 +0000
Patch-mainline: yes
From: olh@suse.de


ChangeSet 1.1994, 2005/01/04 20:15:00-08:00, rusty@rustcorp.com.au

	[PATCH] Fix for NAT core on nonlinear skbs
	
	The nat core calls skb_ip_make_writable() with a length too short.
	Found by nfsim.
	
	Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
	Signed-off-by: Linus Torvalds <torvalds@osdl.org>



 ip_nat_core.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)


diff -Nru a/net/ipv4/netfilter/ip_nat_core.c b/net/ipv4/netfilter/ip_nat_core.c
--- a/net/ipv4/netfilter/ip_nat_core.c	2005-01-04 21:06:46 -08:00
+++ b/net/ipv4/netfilter/ip_nat_core.c	2005-01-04 21:06:46 -08:00
@@ -466,7 +466,7 @@
 	struct iphdr *iph;
 
 	(*pskb)->nfcache |= NFC_ALTERED;
-	if (!skb_ip_make_writable(pskb, iphdroff+sizeof(iph)))
+	if (!skb_ip_make_writable(pskb, iphdroff+sizeof(*iph)))
 		return 0;
 
 	iph = (void *)(*pskb)->data + iphdroff;
-
To unsubscribe from this list: send the line "unsubscribe bk-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

